src/main/webapp/WEB-INF/view/admin.jsp

```diff
@@ -0,0 +1,82 @@
+<%--
+  Copyright 2017 Google Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--%>
+<%-- Imports needed for Admin Page --%>
+
+<%-- Number of users --%>
+<%@ page import="codeu.model.data.User" %>
```

Comment:
It is best practice to keep logic out of the view, but rather in the controller. Let's move all of this logic to the AdminServlet and just pass the values to the view as an attribute on the get request. See here for an example: https://github.com/summerwinter7/CodeU-Summer-2018/blob/d8c9d9799d038110962a2ee24f92f11621375da3/src/main/java/codeu/controller/ActivityfeedServlet.java#L92

Reply:
When I move the logic into AdminServlet and run the program, I get a NullPointerException.